linera-chain/src/data_types.rs

```diff
@@ -636,8 +687,8 @@ impl Certificate {
     /// Returns whether the validator is among the signatories of this certificate.
     pub fn is_signed_by(&self, validator_name: &ValidatorName) -> bool {
         self.signatures
-            .iter()
-            .any(|(name, _)| name == validator_name)
+            .binary_search_by(|(name, _)| name.cmp(validator_name))
```

Comment:
```suggestion
            .binary_search(validator_name)
```

(That should be equivalent, I think?)

Reply:
I unfortunately can't do it like that because it'll still look at both elements of the tuple if I do it like that, and the `Ord` trait isn't implemented for `Signature`, so it doesn't compile (and I don't think it's worth to implement that trait for `Signature` because of just this use case)